-1

Tested at: 1b5a8a4ba8b71af0ebf5501777e24987bad80f6d

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17248/17379/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1TauRecoTreeProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1Muon2RecoTreeProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1MetFilterRecoTreeProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1ExtraTreeProducer.cc 
/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_0_X_2017-01-22-2300/external/slc6_amd64_gcc530/lib/libCore.so: error adding symbols: File in wrong format
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc6_amd64_gcc530/src/L1Trigger/L1TCalorimeter/src/L1TriggerL1TCalorimeter/libL1TriggerL1TCalorimeter.so] Error 1
Leaving library rule at L1Trigger/L1TCalorimeter
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1ElectronRecoTreeProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1UpgradeTreeProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-22-2300/src/L1Trigger/L1TNtuples/plugins/L1uGTTreeProducer.cc 
</pre>

--- a/CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc
+++ b/CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc
@@ -238,126 +238,119 @@ void HcaluLUTTPGCoder::update(const HcalDbService& conditions) {
     cosh_ieta[i] = cosh((eta1 + eta2)/2.);
   }
 
-  HcalSubdetector subdets[] = {HcalBarrel, HcalEndcap, HcalForward};
-  for (int isub = 0; isub < 3; ++isub){
-    HcalSubdetector subdet = subdets[isub];
-    for (int ieta = -HcalDetId::kHcalEtaMask2; ieta <= HcalDetId::kHcalEtaMask2; ++ieta) {
-      for (int iphi = 0; iphi <= HcalDetId::kHcalPhiMask2; ++iphi) {
-	for (int depth = 1; depth < HcalDetId::kHcalDepthMask2; ++depth) {
-	  HcalDetId cell(subdet, ieta, iphi, depth);
-	  if (!topo_->valid(cell)) continue;
+  for (const auto& id: metadata->getAllChannels()) {
+     HcalDetId cell(id);
+     if (!topo_->valid(cell))
+        continue;
+     HcalSubdetector subdet = cell.subdet();
 
-	  const HcalQIECoder* channelCoder = conditions.getHcalCoder (cell);
-	  const HcalQIEShape* shape = conditions.getHcalShape(cell);
-	  HcalCoderDb coder (*channelCoder, *shape);
-	  const HcalLutMetadatum *meta = metadata->getValues(cell);
+     const HcalQIECoder* channelCoder = conditions.getHcalCoder (cell);
+     const HcalQIEShape* shape = conditions.getHcalShape(cell);
+     HcalCoderDb coder (*channelCoder, *shape);
+     const HcalLutMetadatum *meta = metadata->getValues(cell);
 
-          unsigned int mipMax = 0;
-          unsigned int mipMin = 0;
+     unsigned int mipMax = 0;
+     unsigned int mipMin = 0;
 
-          if (topo_->triggerMode() >= HcalTopologyMode::TriggerMode_2017) {
-             const HcalTPChannelParameter *channelParameters = conditions.getHcalTPChannelParameter(cell);
-             mipMax = channelParameters->getFGBitInfo() >> 16;
-             mipMin = channelParameters->getFGBitInfo() & 0xFFFF;
-          }
+     if (topo_->triggerMode() >= HcalTopologyMode::TriggerMode_2017) {
+        const HcalTPChannelParameter *channelParameters = conditions.getHcalTPChannelParameter(cell);
+        mipMax = channelParameters->getFGBitInfo() >> 16;
+        mipMin = channelParameters->getFGBitInfo() & 0xFFFF;
+     }
 
-	  int lutId = getLUTId(subdet, ieta, iphi, depth);
-	  float ped = 0;
-	  float gain = 0;
-	  uint32_t status = 0;
+     int lutId = getLUTId(cell);
+     float ped = 0;
+     float gain = 0;
+     uint32_t status = 0;
 
-	  if (LUTGenerationMode_){
-	    const HcalCalibrations& calibrations = conditions.getHcalCalibrations(cell);
-	    for (int capId = 0; capId < 4; ++capId){
-	      ped += calibrations.pedestal(capId);
-	      gain += calibrations.LUTrespcorrgain(capId);
-	    }
-	    ped /= 4.0;
-	    gain /= 4.0;
+     if (LUTGenerationMode_){
+        const HcalCalibrations& calibrations = conditions.getHcalCalibrations(cell);
+        for (int capId = 0; capId < 4; ++capId){
+           ped += calibrations.pedestal(capId);
+           gain += calibrations.LUTrespcorrgain(capId);
+        }
+        ped /= 4.0;
+        gain /= 4.0;
 
-	    //Get Channel Quality
-	    const HcalChannelStatus* channelStatus = conditions.getHcalChannelStatus(cell);
-	    status = channelStatus->getValue();
-	  } else {
-	    const HcalL1TriggerObject* myL1TObj = conditions.getHcalL1TriggerObject(cell);
-	    ped = myL1TObj->getPedestal();
-	    gain = myL1TObj->getRespGain();
-	    status = myL1TObj->getFlag();
-	  } // LUTGenerationMode_
+        //Get Channel Quality
+        const HcalChannelStatus* channelStatus = conditions.getHcalChannelStatus(cell);
+        status = channelStatus->getValue();
+     } else {
+        const HcalL1TriggerObject* myL1TObj = conditions.getHcalL1TriggerObject(cell);
+        ped = myL1TObj->getPedestal();
+        gain = myL1TObj->getRespGain();
+        status = myL1TObj->getFlag();
+     } // LUTGenerationMode_
 
-	  ped_[lutId] = ped;
-	  gain_[lutId] = gain;
-	  bool isMasked = ( (status & bitToMask_) > 0 );
-	  float rcalib = meta->getRCalib();
+     ped_[lutId] = ped;
+     gain_[lutId] = gain;
+     bool isMasked = ( (status & bitToMask_) > 0 );
+     float rcalib = meta->getRCalib();
 
-	  // Input LUT for HB/HE/HF
-	  if (subdet == HcalBarrel || subdet == HcalEndcap){
-	    HBHEDataFrame frame(cell);
-	    frame.setSize(1);
-	    CaloSamples samples(cell, 1);
+     // Input LUT for HB/HE/HF
+     if (subdet == HcalBarrel || subdet == HcalEndcap){
+        HBHEDataFrame frame(cell);
+        frame.setSize(1);
+        CaloSamples samples(cell, 1);
 
-	    int granularity = meta->getLutGranularity();
+        int granularity = meta->getLutGranularity();
 
-	    for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
-	      frame.setSample(0,HcalQIESample(adc));
-	      coder.adc2fC(frame,samples);
-	      float adc2fC = samples[0];
+        for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
+           frame.setSample(0,HcalQIESample(adc));
+           coder.adc2fC(frame,samples);
+           float adc2fC = samples[0];
 
-	      if (isMasked) inputLUT_[lutId][adc] = 0;
-	      else inputLUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE8_LUT_BITMASK);
-	    }
+           if (isMasked) inputLUT_[lutId][adc] = 0;
+           else inputLUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE8_LUT_BITMASK);
+        }
 
-            unsigned short data[] = {0, 0, 0};
-            QIE11DataFrame upgradeFrame(edm::DataFrame(0, data, 3));
-            CaloSamples upgradeSamples(cell, 1);
-            for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
-               upgradeFrame.setSample(0, adc, 0, true);
-               coder.adc2fC(upgradeFrame, upgradeSamples);
-               float adc2fC = upgradeSamples[0];
+        unsigned short data[] = {0, 0, 0};
+        QIE11DataFrame upgradeFrame(edm::DataFrame(0, data, 3));
+        CaloSamples upgradeSamples(cell, 1);
+        for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
+           upgradeFrame.setSample(0, adc, 0, true);
+           coder.adc2fC(upgradeFrame, upgradeSamples);
+           float adc2fC = upgradeSamples[0];
 
-               if (isMasked) {
-                  upgradeQIE11LUT_[lutId][adc] = 0;
-               } else {
-                  upgradeQIE11LUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE11_LUT_BITMASK);
-                  if (adc >= mipMin and adc < mipMax)
-                     upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB0;
-                  else if (adc >= mipMax)
-                     upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB1;
-               }
-            }
-	  }  // endif HBHE
-	  else if (subdet == HcalForward){
-             HFDataFrame frame(cell);
-             frame.setSize(1);
-             CaloSamples samples(cell, 1);
+           if (isMasked) {
+              upgradeQIE11LUT_[lutId][adc] = 0;
+           } else {
+              upgradeQIE11LUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE11_LUT_BITMASK);
+              if (adc >= mipMin and adc < mipMax)
+                 upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB0;
+              else if (adc >= mipMax)
+                 upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB1;
+           }
+        }
+     }  // endif HBHE
+     else if (subdet == HcalForward){
+        HFDataFrame frame(cell);
+        frame.setSize(1);
+        CaloSamples samples(cell, 1);
 
-             for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
-                frame.setSample(0,HcalQIESample(adc));
-                coder.adc2fC(frame,samples);
-                float adc2fC = samples[0];
-                if (isMasked) inputLUT_[lutId][adc] = 0;
-                else inputLUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[abs(ieta)] )), QIE8_LUT_BITMASK);
-             }
+        for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
+           frame.setSample(0,HcalQIESample(adc));
+           coder.adc2fC(frame,samples);
+           float adc2fC = samples[0];
+           if (isMasked) inputLUT_[lutId][adc] = 0;
+           else inputLUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[cell.ietaAbs()] )), QIE8_LUT_BITMASK);
+        }
 
-             unsigned short data[] = {0, 0, 0, 0};
-             QIE10DataFrame upgradeFrame(edm::DataFrame(0, data, 4));
-             CaloSamples upgradeSamples(cell, 1);
-             for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
-                upgradeFrame.setSample(0, adc, 0, 0, 0, true);
-                coder.adc2fC(upgradeFrame, upgradeSamples);
-                float adc2fC = upgradeSamples[0];
+        unsigned short data[] = {0, 0, 0, 0};
+        QIE10DataFrame upgradeFrame(edm::DataFrame(0, data, 4));
+        CaloSamples upgradeSamples(cell, 1);
+        for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
+           upgradeFrame.setSample(0, adc, 0, 0, 0, true);
+           coder.adc2fC(upgradeFrame, upgradeSamples);
+           float adc2fC = upgradeSamples[0];
 
-                if (isMasked)
-                   upgradeQIE10LUT_[lutId][adc] = 0;
-                else
-                   upgradeQIE10LUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[abs(ieta)] )), QIE10_LUT_BITMASK);
-             }
-	  } // endif HF
-
-	} // for depth
-      } // for iphi
-    } // for iphi
-  }// for subdet
+           if (isMasked)
+              upgradeQIE10LUT_[lutId][adc] = 0;
+           else
+              upgradeQIE10LUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[cell.ietaAbs()] )), QIE10_LUT_BITMASK);
+        }
+     } // endif HF
+  }// for cell
 }
 
 void HcaluLUTTPGCoder::adc2Linear(const HBHEDataFrame& df, IntegerCaloSamples& ics) const {

--- a/L1Trigger/Configuration/python/L1TDigiToRaw_cff.py
+++ b/L1Trigger/Configuration/python/L1TDigiToRaw_cff.py
@@ -15,11 +15,12 @@ from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
 stage2L1Trigger.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("caloStage2Raw")) )
 stage2L1Trigger.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("gmtStage2Raw")) )
 stage2L1Trigger.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("gtStage2Raw")) )
+from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
 
 #
 # Legacy Trigger:
 #
-if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen()):
+if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen() or phase2_common.isChosen()):
     print "L1TDigiToRaw Sequence configured for Run1 (Legacy) trigger. "
     # legacy L1 packages:
     from EventFilter.CSCTFRawToDigi.csctfpacker_cfi import *
@@ -40,7 +41,7 @@ if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen()):
 #
 # Stage-1 Trigger
 #
-if stage1L1Trigger.isChosen() and not stage2L1Trigger.isChosen():
+if stage1L1Trigger.isChosen() and not (stage2L1Trigger.isChosen() or phase2_common.isChosen()):
     print "L1TDigiToRaw Sequence configured for Stage-1 (2015) trigger. "    
     # legacy L1 packers, still in use for 2015:
     from EventFilter.CSCTFRawToDigi.csctfpacker_cfi import *
@@ -65,7 +66,7 @@ if stage1L1Trigger.isChosen() and not stage2L1Trigger.isChosen():
 #
 # Stage-2 Trigger
 #
-if stage2L1Trigger.isChosen():
+if (stage2L1Trigger.isChosen() or phase2_common.isChosen()):
     print "L1TDigiToRaw Sequence configured for Stage-2 (2016) trigger. "
     from EventFilter.L1TRawToDigi.caloStage2Raw_cfi import *
     from EventFilter.L1TRawToDigi.gmtStage2Raw_cfi import *

--- a/L1Trigger/Configuration/python/L1TRawToDigi_cff.py
+++ b/L1Trigger/Configuration/python/L1TRawToDigi_cff.py
@@ -69,6 +69,7 @@ def unpack_stage2():
 #
 from Configuration.Eras.Modifier_stage1L1Trigger_cff import stage1L1Trigger
 from Configuration.Eras.Modifier_stage2L1Trigger_cff import stage2L1Trigger
+from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
 if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen()):
     print "L1TRawToDigi Sequence configured for Run1 (Legacy) trigger. "
     unpack_legacy()
@@ -77,7 +78,7 @@ if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen()):
 #
 # Stage-1 Trigger
 #
-if stage1L1Trigger.isChosen() and not stage2L1Trigger.isChosen():
+if stage1L1Trigger.isChosen() and not (stage2L1Trigger.isChosen() or phase2_common.isChosen()):
     print "L1TRawToDigi Sequence configured for Stage-1 (2015) trigger. "    
     unpack_stage1()
     L1TRawToDigi = cms.Sequence(L1TRawToDigi_Stage1)
@@ -85,7 +86,7 @@ if stage1L1Trigger.isChosen() and not stage2L1Trigger.isChosen():
 #
 # Stage-2 Trigger:  fow now, unpack Stage 1 and Stage 2 (in case both available)
 #
-if stage2L1Trigger.isChosen():
+if (stage2L1Trigger.isChosen() or phase2_common.isChosen()):
     print "L1TRawToDigi Sequence configured for Stage-2 (2016) trigger. "    
     unpack_stage1()
     unpack_stage2()

--- a/SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py
+++ b/SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py
@@ -9,7 +9,8 @@ hcalDigiSequence = cms.Sequence(simHcalTriggerPrimitiveDigis
                                 +simHcalDigis
                                 *simHcalTTPDigis)
 
-_phase2_hcalDigiSequence = hcalDigiSequence.copyAndExclude([simHcalTriggerPrimitiveDigis,simHcalTTPDigis])
+#_phase2_hcalDigiSequence = hcalDigiSequence.copyAndExclude([simHcalTriggerPrimitiveDigis,simHcalTTPDigis])
+#_phase2_hcalDigiSequence = hcalDigiSequence.copyAndExclude([simHcalTTPDigis])
 
-from Configuration.Eras.Modifier_phase2_hcal_cff import phase2_hcal
-phase2_hcal.toReplaceWith( hcalDigiSequence, _phase2_hcalDigiSequence )
+#from Configuration.Eras.Modifier_phase2_hcal_cff import phase2_hcal
+#phase2_hcal.toReplaceWith( hcalDigiSequence, _phase2_hcalDigiSequence )

